src/com/cloudogu/ces/cesbuildlib/SCMManager.groovy

```diff
@@ -0,0 +1,125 @@
+package com.cloudogu.ces.cesbuildlib
+
+class SCMManager implements Serializable{
+
+  private script
+  private credentials
+  Sh sh
+  String repositoryUrl
+
+  SCMManager(script, credentials) {
+    this.script = script
+    this.sh = new Sh(script)
+    this.credentials = credentials
+  }
+
+  protected String executeWithCredentials(Closure closure) {
+    if (credentials) {
+      script.withCredentials([script.usernamePassword(credentialsId: credentials,
+          passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USER')]) {
+        closure.call(true)
+      }
+    } else {
+      closure.call(false)
+    }
+  }
+
+  protected String getCurlAuthParam() {
+    String curlAuthParam = ""
+    if(credentials) {
+      executeWithCredentials {
+        curlAuthParam = "-u ${script.env.GIT_USER}:${script.env.GIT_PASSWORD}"
+      }
+    }
+    return curlAuthParam
+  }
+
+  Object searchPullRequestByTitle(String title) {
+    def pullRequest
+    for (Map pr : getPullRequests()) {
+      if (pr["title"] == title) {
+        pullRequest = pr
+      }
+    }
+
+    return pullRequest
+  }
+
+  Object getPullRequests() {
+    String curlCommand = """curl -w "%{http_code}" ${getCurlAuthParam()} -H 'Content-Type: application/vnd.scmm-pullRequestCollection+json;v=2' https://${this.repositoryUrl}"""
+    String httpResponse = script.sh returnStdout: true, script: curlCommand
+
+    //get only the http code (3 characters) of the response.
+    String httpCode = httpResponse.reverse().take(3).reverse()
+    //drop the http code (last 3 characters) of this string
+    String pullRequests = httpResponse.reverse().drop(3).reverse()
+
+    httpCode = httpCode.trim()
+    script.echo "Getting all pull requests yields httpCode: ${httpCode}"
+    if (httpCode != "200") {
+      script.unstable 'Could not create pull request'
+    }
+
+    def prsAsJson = script.readJSON text: pullRequests
```

Comment:
This step is part of the pipeline utilities plugin. Either we use something else or we need to document this as a requirement

Reply:
Fixed in f7846b7e